stackit/internal/services/argus/instance/resource.go

```diff
@@ -784,6 +1092,174 @@ func mapMetricsRetentionField(r *argus.GetMetricsStorageRetentionResponse, model
+	}
+
+	// Receivers
+	respReceivers := *resp.Data.Receivers
+	receiversList := []attr.Value{}
+
+	for i := range respReceivers {
+		receiver := respReceivers[i]
+
+		// emailConfigs
+		emailConfigList := []attr.Value{}
+		if receiver.EmailConfigs != nil {
+			for _, emailConfig := range *receiver.EmailConfigs {
+				emailConfigList = append(emailConfigList, types.ObjectValueMust(emailConfigsTypes, map[string]attr.Value{
+					"auth_identity": types.StringPointerValue(emailConfig.AuthIdentity),
+					"auth_password": types.StringPointerValue(emailConfig.AuthPassword),
+					"auth_username": types.StringPointerValue(emailConfig.AuthUsername),
+					"from":          types.StringPointerValue(emailConfig.From),
+					"smart_host":    types.StringPointerValue(emailConfig.Smarthost),
+					"to":            types.StringPointerValue(emailConfig.To),
+				}))
+			}
+		}
+
+		// opsgenieConfigs
+		opsgenieConfigList := []attr.Value{}
+		if receiver.OpsgenieConfigs != nil {
+			for _, opsgenieConfig := range *receiver.OpsgenieConfigs {
+				opsgenieConfigList = append(opsgenieConfigList, types.ObjectValueMust(opsgenieConfigsTypes, map[string]attr.Value{
+					"api_key": types.StringPointerValue(opsgenieConfig.ApiKey),
+					"api_url": types.StringPointerValue(opsgenieConfig.ApiUrl),
+					"tags":    types.StringPointerValue(opsgenieConfig.Tags),
+				}))
+			}
+		}
+
+		// webhooksConfigs
+		webhooksConfigList := []attr.Value{}
+		if receiver.WebHookConfigs != nil {
+			for _, webhookConfig := range *receiver.WebHookConfigs {
+				webhooksConfigList = append(webhooksConfigList, types.ObjectValueMust(webHooksConfigsTypes, map[string]attr.Value{
+					"url":      types.StringPointerValue(webhookConfig.Url),
+					"ms_teams": types.BoolPointerValue(webhookConfig.MsTeams),
+				}))
+			}
+		}
+
+		receiversList = append(receiversList, types.ObjectValueMust(receiversTypes, map[string]attr.Value{
+			"name":             types.StringPointerValue(receiver.Name),
+			"email_configs":    types.ListValueMust(types.ObjectType{AttrTypes: emailConfigsTypes}, emailConfigList),
+			"opsgenie_configs": types.ListValueMust(types.ObjectType{AttrTypes: opsgenieConfigsTypes}, opsgenieConfigList),
+			"webhooks_configs": types.ListValueMust(types.ObjectType{AttrTypes: webHooksConfigsTypes}, webhooksConfigList),
+		}))
+	}
+
+	alertConfig := types.ObjectValueMust(alertConfigTypes, map[string]attr.Value{
+		"receivers": types.ListValueMust(types.ObjectType{AttrTypes: receiversTypes}, receiversList),
+		"route": types.ObjectValueMust(routeTypes, map[string]attr.Value{
+			"receiver": types.StringPointerValue(resp.Data.Route.Receiver),
+		}),
+	})
+
+	modelMockAlertConfig, diags := types.ObjectValueFrom(ctx, alertConfigTypes, mockAlertConfig)
+	if diags.HasError() {
+		return fmt.Errorf("mapping mock alert config: %w", core.DiagsToError(diags))
+	}
+
+	if alertConfig.Equal(modelMockAlertConfig) {
+		model.AlertConfig = types.ObjectNull(alertConfigTypes)
+		return nil
+	}
```

Comment:
This should also have a comment explaining the why behind it

Reply:
Agreed, I added a comment